aiohttp/cookiejar.py

```diff
@@ -202,7 +203,10 @@ def filter_cookies(self, request_url: URL=URL()) -> 'BaseCookie[str]':
                           .format(type(request_url)),
                           DeprecationWarning)
             request_url = URL(request_url)
-        filtered = SimpleCookie()  # type: SimpleCookie[str]
+        if self._quote_cookie:
+            filtered = SimpleCookie()  # type: SimpleCookie[str]
+        else:
+            filtered = BaseCookie()  # type: BaseCookie[str]
```

Comment:
I see that this isn't allowed in MyPy, but I'm not sure how you'd like to look, so I'll await your feedback.

Something like this could work:

```py
filtered = SimpleCookie() if self._quote_cookie else BaseCookie()  # type: Union[SimpleCookie[str], BaseCookie[str]] # noqa
```